subs/pantry/src/Pantry.hs

```diff
@@ -510,7 +512,9 @@ loadCabalFileBytes (PLIHackage pir _mtree) = getHackageCabalFile pir
 loadCabalFileBytes pl = do
   package <- loadPackage pl
   let sfp = cabalFileName $ pkgName $ packageIdent package
-      TreeEntry cabalBlobKey _ft = packageCabalEntry package
+  cabalBlobKey <- case (packageCabalEntry package) of
+                       PCHpack _ -> throwIO $ TreeWithoutCabalFile pl
```

Comment:
This doesn't feel right. Why aren't we generating the cabal file in this case? There's no indication at the type level or the docs that this may fail. At the very least, I'd say we should return a `Maybe` value instead of using runtime exceptions, since this is an expected case.

Reply:
This is fixed.